### PR TITLE
feat: add ftp server support for backup replay fetch method; add missing requirements

### PR DESCRIPTION
**Pull request by cmyui**

### CODE REVIEW

File: app/api/replays.py [hunk 1]
```diff
@@ -34,6 +34,20 @@ async def get_replay(
         f"http://localhost:3030/get?id={score_id}",
     ) as session:
         if not session or session.status != 200:
+            try:
+                stream = await app.state.services.ftp_client.download_stream(
+                    source=f"/replays/replay_{score_id}.osr",
+                )
+                replay_data = await stream.read()
+                print(len(replay_data), "replay bytes from ftp")
```
Comment (tsunyoku) [inline app/api/replays.py]: dbg print

File: app/api/replays.py [hunk 2]
```diff
@@ -34,6 +34,20 @@ async def get_replay(
...
+                return replay_data
+            except Exception as e:
+                import traceback
+
+                print("\n\nYEAAA")
```
Comment (tsunyoku) [inline app/api/replays.py]: dbg print

File: app/api/replays.py [hunk 3]
```diff
@@ -34,6 +34,20 @@ async def get_replay(
...
+                traceback.print_exc()
+                pass
```
Comment (tsunyoku) [inline app/api/replays.py]: pass is unnecessary here

**Review by tsunyoku — CHANGES_REQUESTED** [review]

**Comment by cmyui** [comment]
haha @tsunyoku im testing no worries lol

**Comment by tsunyoku** [comment]
sjfjenfkwnfn

**Comment by ch03e** [comment]
hello discord webhook!!!!!!

**Review by tsunyoku — APPROVED** [review]

**Pull request merged**